File: server/src/user/vo/user.vo.ts
import { UserEntity } from '../../database/entities/user.entity';

export class UserVo {
  id: string;
  userAccount: string;
  username: string;
  userAvatar: string;
  userProfile: string;
  userRole: string;
  createTime: Date;

  static fromEntity(entity: UserEntity): UserVo | null {
    if (!entity) return null;
    const vo = new UserVo();
    vo.id = String(entity.id);
    vo.userAccount = entity.userAccount;
    vo.username = entity.username;
    vo.userAvatar = entity.userAvatar;
    vo.userProfile = entity.userProfile;
    vo.userRole = entity.userRole;
    vo.createTime = entity.createTime;
    return vo;
  }
}

export class LoginUserVo {
  id: string;
  userAccount: string;
  username: string;
  userAvatar: string;
  userProfile: string;
  userRole: string;

  static fromEntity(entity: UserEntity): LoginUserVo | null {
    if (!entity) return null;
    const vo = new LoginUserVo();
    vo.id = String(entity.id);
    vo.userAccount = entity.userAccount;
    vo.username = entity.username;
    vo.userAvatar = entity.userAvatar;
    vo.userProfile = entity.userProfile;
    vo.userRole = entity.userRole;
    return vo;
  }
}

File: src/main/java/com/github/tianchenghang/model/vo/UserVo.java
package com.github.tianchenghang.model.vo;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;
import lombok.Data;

@Data
public class UserVo implements Serializable {

  private Long id;

  private String userAccount;

  private String username;

  private String userAvatar;

  private String userProfile;

  private String userRole;

  private LocalDateTime createTime;

  @Serial private static final long serialVersionUID = 1L;
}
